src/key_gen/encryptor.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use super::Error;
+use crate::id::PublicId;
+use aes::Aes128;
+use block_modes::block_padding::Pkcs7;
+use block_modes::{BlockMode, Cbc};
+use rand::Rng;
+use serde_derive::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+
+type Aes128Cbc = Cbc<Aes128, Pkcs7>;
```

Comment:
So we are using symmetric keys now. Does that mean that the stuff we encrypt is supposed to be only decrypted by us again?

Reply:
no. the encrypted stuff will be recorded by others. And later on, when we are complained, we need to send them the encryption key so that they can decrypt the stuff and validate whether we are really invalid.